cosmos/operators/watcher.py

```diff
@@ -50,13 +53,25 @@
     EventMsg = None
 
 logger = logging.getLogger(__name__)
-
+xcom_set_lock = Lock()
 
 CONSUMER_OPERATOR_DEFAULT_PRIORITY_WEIGHT = 10
 PRODUCER_OPERATOR_DEFAULT_PRIORITY_WEIGHT = 9999
 WEIGHT_RULE = "absolute"  # the default "downstream" does not work with dag.test()
 
 
+def safe_xcom_push(task_instance: TaskInstance, key: str, value: Any) -> None:
+    """
+    Safely set an XCom value in a thread-safe manner in Airflow 3.0 and 3.1.
```

Comment:
Do we need to mention specific Airflow versions here? Because we are not altering the behavior here based on Airflow versions here.